### ISSUE TITLE
Handle local private key in omni-configure for portal

### ISSUE THREAD

**Issue by tcmitchell** [GENI-NSF/geni-tools#267]
omni-configure fails if a portal experimenter uploaded a certificate signing request (CSR) instead of having the portal generate a private key. omni-configure should handle this case, which can be detected by the lack of a private key in the omni-bundle.zip file.

This may be simply putting a placeholder in the generated omni config file.

 _Imported from trac ticket [#267](http://trac.gpolab.bbn.com/gcf/ticket/267),  created by tmitchel on 05-16-2013 at 13:56, last modified: 05-30-2013 at 08:13_

**Comment by innnkkki** [comment]
This is fixed in commit [fb4cf62f9de9df792fd3b912d0a3256b1e41b2d4](https://github.com/GENI-NSF/geni-tools/commit/fb4cf62f9de9df792fd3b912d0a3256b1e41b2d4). 

Changes that happened:
  * repurposed the -k option that was for specifying the private key for the PL cert to be generic for any CH that needs to have a separate cert key that is not in the cert file. Changed its long name from --plkey to --prcertkey
  * the order in which it is decided which key to use for the ssl cert is:
    1) check the cert file to see if there is a key there, if there is use this and ignore any other option
    2) check if the user has provided a location through the -k option
    3) if not use the default location, which is:
        * geni_key_pg for PG CH
        * geni_key_pl for PL CH
        * geni_ssl_porta.key for portal CH

This fix ended up with extensive code restructuring so all cases need to be tested. 

I have tested with a portal account that has a cert with all three options, with my pgeni and my PLC accounts. The testing was minimal. 


_Trac comment by nriga (github user: innnkkki) on 05-23-2013 at 18:00_

**Comment by luisanevers** [comment]
Verified local keys for "portal" and "PG" framework, but ran into a problem with PlanetLab keys with both my PI and regular user keys:
```
lnevers@sendaria:$ omni-configure.py -f pl -p ~/.ssl/planetlab/lnevers.pkey 
INFO:omniconfig:Using configfile: /home/lnevers/.gcf/omni_config
INFO:omniconfig:Using certfile /home/lnevers/.ssl/planetlab/lnevers.pkey
INFO:omniconfig:Using PlanteLab private key file /home/lnevers/.ssl/planetlab/lnevers.pkey
INFO:omniconfig:

	CREATING SSH KEYPAIR
INFO:omniconfig:Private key stored at: /home/lnevers/.ssh/geni_key_pl
INFO:omniconfig:Public key stored at: /home/lnevers/.ssh/geni_key_pl.pub
Traceback (most recent call last):
  File "/home/lnevers/gcf-2.3-rc2/src/omni-configure.py", line 1196, in <module>
    sys.exit(main())
  File "/home/lnevers/gcf-2.3-rc2/src/omni-configure.py", line 1190, in main
    createConfigFile(opts,pub_key_file_list)
  File "/home/lnevers/gcf-2.3-rc2/src/omni-configure.py", line 742, in createConfigFile
    cert = Certificate(filename=opts.cert)
  File "/home/lnevers/gcf-2.3-rc2/src/sfa/trust/certificate.py", line 330, in __init__
    self.load_from_file(filename)
  File "/home/lnevers/gcf-2.3-rc2/src/sfa/trust/certificate.py", line 397, in load_from_file
    self.load_from_string(string)
  File "/home/lnevers/gcf-2.3-rc2/src/sfa/trust/certificate.py", line 383, in load_from_string
    self.cert = crypto.load_certificate(crypto.FILETYPE_PEM, parts[0])
OpenSSL.crypto.Error: [routines', 'PEM_read_bio', 'bad end line')](('PEM)
```

PI Keys:
```
lnevers@sendaria:$ omni-configure.py -f pl -p ~/.gcf/luisa_nevers.pkey      
INFO:omniconfig:Using configfile: /home/lnevers/.gcf/omni_config
INFO:omniconfig:Using certfile /home/lnevers/.gcf/luisa_nevers.pkey
INFO:omniconfig:Using PlanteLab private key file /home/lnevers/.gcf/luisa_nevers.pkey
INFO:omniconfig:

	CREATING SSH KEYPAIR
INFO:omniconfig:Loading SSH key from /home/lnevers/.gcf/luisa_nevers.pkey
INFO:omniconfig:File /home/lnevers/.ssh/geni_key_pl already exists. Loading SSH key from /home/lnevers/.ssh/geni_key_pl
INFO:omniconfig:Private key stored at: /home/lnevers/.ssh/geni_key_pl
INFO:omniconfig:Public key stored at: /home/lnevers/.ssh/geni_key_pl.pub
Traceback (most recent call last):
  File "/home/lnevers/gcf-2.3-rc2/src/omni-configure.py", line 1196, in <module>
    sys.exit(main())
  File "/home/lnevers/gcf-2.3-rc2/src/omni-configure.py", line 1190, in main
    createConfigFile(opts,pub_key_file_list)
  File "/home/lnevers/gcf-2.3-rc2/src/omni-configure.py", line 742, in createConfigFile
    cert = Certificate(filename=opts.cert)
  File "/home/lnevers/gcf-2.3-rc2/src/sfa/trust/certificate.py", line 330, in __init__
    self.load_from_file(filename)
  File "/home/lnevers/gcf-2.3-rc2/src/sfa/trust/certificate.py", line 397, in load_from_file
    self.load_from_string(string)
  File "/home/lnevers/gcf-2.3-rc2/src/sfa/trust/certificate.py", line 383, in load_from_string
    self.cert = crypto.load_certificate(crypto.FILETYPE_PEM, parts[0])
OpenSSL.crypto.Error: [routines', 'PEM_read_bio', 'bad end line')](('PEM)
lnevers@sendaria:$ 
```


_Trac comment by lnevers (github user: luisanevers) on 05-30-2013 at 08:13_
